### DISCUSSION TITLE
Stupid question : Birdnet-Pi on a smartphone ?

### DISCUSSION THREAD

**Discussion by Fish31520** [mcguirepr89/BirdNET-Pi#978] [Q&A] [answered]
Hi folks,

Few days ago my mother gave me her old smartphone. It's working pretty fine but she wanted a new one (I tried to convince her that it was a useless consumerism behavior but she really didn't mind...).
Anyway, now I've got this new "old smartphone" 😄 and I'm wondering if there could be a way to use it as some kind of RPi (it already has a wifi connection, a microphone, a processor so we mustn't be far from what a RPi actually is).

So here comes my newb's question 😄 : Is there a way to get rid of the Android OS and maybe to make it boot on a MicroSD card with the right OS installed ?

Thanks for your help !

Jean

**Comment by jflyons** [comment, marked as answer]
Hi Jean - I have been running the Birdweather app on an old iPad, and if your device is iOS I would encourage you to try it. I have it covering a different part of my patch as compared to my BirdNET-Pi. Not sure how their detection models align but both are based on BirdNET, and their results - when it's a bird heard in both microphone locations - quite consistent.

**Comment by Fish31520** [comment]
@jflyons 
Hi,

Thanks for your help and sorry for that late answer.

I've search a bit and it seems like it's super hard (almost impossible ?) to install a Linux OS on an old smartphone.

The IOS idea is good but unfortunately I only have Android smartphones to play with.

Anyway, thanks for your help.

**Comment by srd424** [comment]
Running a fully native Linux on a random old Android phone would likely be a major effort - you'd likely need proprietary device drivers and lots of messing around unlocking bootloaders etc. Having said that, projects like postmarketOS seem to be able to do that on a specific set of devices: https://wiki.postmarketos.org/wiki/Devices

Running a Linux-like userspace on an unmodified Android firmware might be more feasible though, checkout https://github.com/AndronixApp/AndronixOrigin, https://github.com/termux/termux-app etc - probably need a fair amount of tinkering with the vanilla BirdNET-Pi install though, but more do-able I would think.

The quickest option might be to find an app that would just stream audio from the phone, and then run BirdNET-Pi somewhere else - @MatthewBCooke did the work to get it going on x86-64: [MatthewBCooke/BirdNET-Pi](https://github.com/MatthewBCooke/BirdNET-Pi), and I'm in the process of cobbling together some docker images: https://github.com/mcguirepr89/BirdNET-Pi/discussions/984

**Reply by Fish31520** [reply]
Wow, that's just to much mess for me ! 😄

I thought a smartphone was just a small computer but it clearly seems to be quite different.

Well, my mother's old phone will end up in the recycling bin ! 😄

Anyway, thanks for your answer !

PS : And unfortunately Postmarket OS doesn't seem to be available for that old phone. Too bad, I may have tried to install it, just for fun.

**Reply by srd424** [reply]
Well, they _are_ small computers, but locked down to a greater or lesser extent, and not necessarily with available drivers :( It's a shame because more stuff could be repurposed like you're describing if manufacturers were a bit more open :(

**Reply by Fish31520** [reply]
Money, money... 😅

**Comment by srd424** [comment]
Trying really hard not to get distracted by trying to port BNPI to Termux .. it looks like microphone recording at least is possible: https://wiki.termux.com/wiki/Termux-microphone-record
